codec/src/main/java/io/netty/handler/codec/compression/ZlibCodecFactory.java

```diff
@@ -33,7 +33,7 @@
     private static final boolean noJdkZlibEncoder;
 
     static {
-        noJdkZlibDecoder = SystemPropertyUtil.getBoolean("io.netty.noJdkZlibDecoder", true);
+        noJdkZlibDecoder = SystemPropertyUtil.getBoolean("io.netty.noJdkZlibDecoder", false);
```

Comment:
Actually we can only use the jdk one if we are using java7 or higher. Because we need this method:

https://docs.oracle.com/javase/7/docs/api/java/util/zip/Deflater.html#deflate(byte[],%20int,%20int,%20int)

So how about change it to:
```java
noJdkZlibDecoder = SystemPropertyUtil.getBoolean("io.netty.noJdkZlibDecoder", PlatformDependent.javaVersion() >= 7);
```